streams/test-utils/src/main/java/org/apache/kafka/streams/test/ConsumerRecordFactory.java

```diff
@@ -0,0 +1,414 @@
+    }
+
+    /**
+     * Create a new factory for the given topic.
+     * Auto-advance is disabled.
+     *
+     * @param keySerializer the key serializer
+     * @param valueSerializer the value serializer
+     * @param startTimestampMs the initial timestamp for generated records
+     */
+    public ConsumerRecordFactory(final Serializer<K> keySerializer,
+                                 final Serializer<V> valueSerializer,
+                                 final long startTimestampMs) {
+        this(null, keySerializer, valueSerializer, startTimestampMs, 0L);
+    }
+
+    /**
+     * Create a new factory for the given topic.
+     * Auto-advance is disabled.
+     *
+     * @param defaultTopicName the topic name used for all generated {@link ConsumerRecord consumer records}
+     * @param keySerializer the key serializer
+     * @param valueSerializer the value serializer
+     * @param startTimestampMs the initial timestamp for generated records
+     */
+    public ConsumerRecordFactory(final String defaultTopicName,
+                                 final Serializer<K> keySerializer,
+                                 final Serializer<V> valueSerializer,
+                                 final long startTimestampMs) {
+        this(defaultTopicName, keySerializer, valueSerializer, startTimestampMs, 0L);
+    }
+
+    /**
+     * Create a new factory for the given topic.
+     *
+     * @param keySerializer the key serializer
+     * @param valueSerializer the value serializer
+     * @param startTimestampMs the initial timestamp for generated records
+     * @param autoAdvanceMs the time increment pre generated record
+     */
+    public ConsumerRecordFactory(final Serializer<K> keySerializer,
+                                 final Serializer<V> valueSerializer,
+                                 final long startTimestampMs,
+                                 final long autoAdvanceMs) {
+        this(null, keySerializer, valueSerializer, startTimestampMs, autoAdvanceMs);
+    }
+
+    /**
+     * Create a new factory for the given topic.
+     *
+     * @param defaultTopicName the topic name used for all generated {@link ConsumerRecord consumer records}
+     * @param keySerializer the key serializer
+     * @param valueSerializer the value serializer
+     * @param startTimestampMs the initial timestamp for generated records
+     * @param autoAdvanceMs the time increment pre generated record
+     */
+    public ConsumerRecordFactory(final String defaultTopicName,
+                                 final Serializer<K> keySerializer,
+                                 final Serializer<V> valueSerializer,
+                                 final long startTimestampMs,
+                                 final long autoAdvanceMs) {
+        Objects.requireNonNull(keySerializer, "keySerializer cannot be null");
+        Objects.requireNonNull(valueSerializer, "valueSerializer cannot be null");
+        this.topicName = defaultTopicName;
```

Comment:
should we also be checking that `topicName` is non-null?

Reply:
I want to allow `topicName=null` for the case that somebody wants a factory without default topic name. We have a constructor without taking a defaultTopicName.